modules/repository/main.tf

```diff
@@ -0,0 +1,69 @@
+resource "github_repository" "this" {
+  name         = var.name
+  description  = local.description
+  homepage_url = local.homepage_url
+
+  visibility             = "public"
+  has_issues             = true
+  has_wiki               = false
+  has_projects           = false
+  allow_merge_commit     = false
+  allow_squash_merge     = true
+  allow_rebase_merge     = false
+  delete_branch_on_merge = true
+  has_downloads          = false
+  archived               = false
+  topics                 = local.topics
+  auto_init              = true
+  license_template       = "apache-2.0"
+
+}
+
+resource "github_branch" "default" {
+  repository = github_repository.this.name
+  branch     = "main"
+}
+
+resource "github_branch_default" "default" {
+  repository = github_repository.this.name
+  branch     = github_branch.default.branch
+}
+
+resource "github_branch_protection" "default" {
+  repository_id = github_repository.this.node_id
+  pattern       = github_branch.default.branch
+
+  # when a repo is being initialized/created you can run into race conditions
+  # by adding an explicit depends we force the repo to be created
+  # before it attempts to add branch protection
+  depends_on = [
+    github_repository.this,
+  ]
+
+  required_status_checks {
+    strict   = true
+    contexts = ["final"]
```

Comment:
This also needs to be configurable, as some status checks don't have final in them. e.g. the terraform ones.

Reply:
Fair point